### PR TITLE
API Update: Notifications & Template Lists

### PR DESCRIPTION
**Pull request by kelunik**
- Respond with full templates in lists
- Add send once in notifications
- Add emails in notifications

### CODE REVIEW

**Comment by kelunik** [comment]
Also @phimonk.

**Comment by simonkorz** [comment]
Changes to `NotificationOperations.java` and `NotificationTransformer.java` are already done in Notification's PR #79

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java [hunk 1]
```diff
@@ -132,31 +156,41 @@ public boolean deleteNotification(int id) {
...
      */
     public boolean updateLastSendForNotification(int notificationID) {
         //TODO? wait for simon!
         return true;//create.update(Tables.NOTIFICATION)
-                //.set(Tables.NOTIFICATION.LASTSENT, Timestamp.valueOf(LocalDateTime.now()))
-                //.where(Tables.NOTIFICATION.ID_NOTIFICATION.eq(notificationID))
-                //.execute() == 1;
+        //.set(Tables.NOTIFICATION.LASTSENT, Timestamp.valueOf(LocalDateTime.now()))
+        //.where(Tables.NOTIFICATION.ID_NOTIFICATION.eq(notificationID))
+        //.execute() == 1;
```
Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java:185]: whats going on here?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java [hunk 2]
```diff
@@ -164,15 +198,42 @@ public Notification updateNotification(int id, Notification notification) {
...
+                        receiver.setNotification(record.getIdNotification());
+                        receiver.setEmail(mail);
+                        return receiver;
+                    })
+                    .collect(Collectors.toList());
+
+            create.transaction(conf -> {
+                DSL.using(conf).deleteFrom(NOTIFICATION_RECEIVER_EMAIL)
+                        .where(NOTIFICATION_RECEIVER_EMAIL.NOTIFICATION.eq(id))
+                        .execute();
```
Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java]: this is not needed...they should delete themself

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java (hunk 1 above)
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java:185]: Autoformat… will be corrected once @phimonk's changes land anyway.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java [hunk 3]
```diff
@@ -96,29 +111,38 @@ public Notification insertNotification(Notification toStore) throws IllegalArgum
                 Notification.NAME_FIELD_NUMBER,
                 Notification.DESCRIPTION_FIELD_NUMBER,
                 Notification.QUERY_FIELD_NUMBER,
-                Notification.CHECK_PERIOD_FIELD_NUMBER);
+                Notification.CHECK_PERIOD_FIELD_NUMBER,
+                Notification.SEND_ONCE_FIELD_NUMBER,
+                Notification.EMAILS_FIELD_NUMBER);
+
+        toStore.getEmailsList().forEach(email -> {
+            if (!EmailValidator.getInstance(false).isValid(email)) {
```
Comment (simonkorz) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java:119]: is this really the right place to validate emails?
i think this should happen when the proto is created and the emails are assigend before one decides to store them.

Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java:119]: We check if all required fields are present as well here, so I think it's the right place.

**Comment by kelunik** [comment]
Rebased @LeanderK.

**Pull request merged**